### PR TITLE
Adding `pylint` 

### PR DESCRIPTION
**Pull request by jamesbraza**
Introduces `pylint` into CI, suppresses a few more errors, and manually fixes all errors that came up

### CODE REVIEW

File: paperqa/settings.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def available_for_inference(self) -> list["ParsingOptions"]:
 def _get_parse_type(opt: ParsingOptions, config: "ParsingSettings") -> str:
     if opt == ParsingOptions.PAPERQA_DEFAULT:
         return config.parser_version_string
-    assert_never()
+    assert_never(opt)
```
Comment (jamesbraza) [inline paperqa/settings.py:69]: @mskarlin fyi it looks like you didn't properly roll out `assert_never`

File: paperqa/agents/__init__.py [hunk 2]
```diff
@@ -79,16 +79,16 @@ def configure_cli_logging(verbosity: int = 0) -> None:
 
     rich_handler.setFormatter(logging.Formatter("%(message)s", datefmt="[%X]"))
 
-    module_logger = logging.getLogger(__name__.split(".")[0])
+    module_logger = logging.getLogger(__name__.split(".", maxsplit=1)[0])
```
Comment (mskarlin) [inline paperqa/agents/__init__.py:82]: I don't think we want maxsplit here -- like `paperqa.agents.search` -- we want two splits there.

File: paperqa/clients/__init__.py [hunk 3]
```diff
@@ -19,19 +19,14 @@
 
 logger = logging.getLogger(__name__)
 
-DEFAULT_CLIENTS: (
```
Comment (mskarlin) [inline paperqa/clients/__init__.py:22]: We're losing some semantics here it takes either a collection OR a sequence of collections -- this is removing the second possibility.

Comment (jamesbraza) [reply paperqa/clients/__init__.py:22]: So the place this gets used still has both. Imo it's fine for the default to only have one of the logic branches

File: paperqa/agents/__init__.py (hunk 2 above)
Comment (jamesbraza) [reply paperqa/agents/__init__.py:82]: We throw away everything beyond `[0]` so I think `maxsplit` would be a slight improvement

File: paperqa/clients/__init__.py (hunk 3 above)
Comment (mskarlin) [reply paperqa/clients/__init__.py:22]: Ahh i see what you're saying I was thinking this was in a doc signature -- makes sense.

**Comment by jamesbraza** [comment]
This is sort of a draft. Some of this is going to go away with `aviary` refactor removing some of the code

**Comment by jamesbraza** [comment]
Okay, it's ready for review

**Review by whitead — APPROVED** [review]
Nice work!

**Review by mskarlin — APPROVED** [review]

**Pull request merged**